core/server/data/meta/context_object.js

```diff
@@ -16,17 +16,24 @@ function getContextObject(data, context) {
     // @TODO: meta layer is very broken, it's really hard to understand what it's doing
     // The problem is that handlebars root object is structured differently. Sometimes the object is flat on data
     // and sometimes the object is part of a key e.g. data.post. This needs to be prepared at the very first stage and not in each helper.
-    if (_.includes(context, 'page') || _.includes(context, 'amp') && data.post) {
+    if ((_.includes(context, 'page') || _.includes(context, 'amp')) && data.post) {
         chosenContext = data.post;
     } else if (_.includes(context, 'post') && data.post) {
         chosenContext = data.post;
+    } else if (_.includes(context, 'page') && data.page) {
+        chosenContext = data.page;
```

Comment:
Added this extra case to ensure the data is returned for the case of using `data: page.slug`.
Dynamic routing has added a lot of extra cases for the meta layer, refs #10082.